### PR TITLE
feat: Open More Endpoints for Customization

### PR DESCRIPTION
**Pull request by danielbankhead**
Completes AL-3

🦕

### CODE REVIEW

File: src/auth/oauth2client.ts [hunk 1]
```diff
@@ -460,64 +529,41 @@ export class OAuth2Client extends AuthClient {
...
-   */
-  private static readonly GOOGLE_OAUTH2_REVOKE_URL_ =
-    'https://oauth2.googleapis.com/revoke';
-
-  /**
-   * Google Sign on certificates in PEM format.
-   */
-  private static readonly GOOGLE_OAUTH2_FEDERATED_SIGNON_PEM_CERTS_URL_ =
-    'https://www.googleapis.com/oauth2/v1/certs';
+    this.endpoints = {
```
Comment (TimurSadykov) [inline src/auth/oauth2client.ts:533]: Do you override those somewhere?

**Review by TimurSadykov — COMMENTED** [review]
questions on endpoint configuration

File: src/auth/oauth2client.ts (hunk 1 above)
Comment (danielbankhead) [reply src/auth/oauth2client.ts:533]: Refactored to use universe domain

File: src/auth/googleauth.ts [hunk 2]
```diff
@@ -1080,24 +1086,24 @@ export class GoogleAuth<T extends AuthClient = JSONClient> {
       throw new Error('Cannot sign data without `client_email`.');
     }
 
-    return this.signBlob(crypto, creds.client_email, data);
+    return this.signBlob(crypto, creds.client_email, data, endpoint);
```
[the remark this answers is not in the document]
Comment (danielbankhead) [reply src/auth/googleauth.ts:1094]: Fixed, default to using universe domain

**Review by alexander-fenster — APPROVED** [review]

File: src/auth/baseexternalclient.ts [hunk 3]
```diff
@@ -53,12 +53,13 @@ export const EXPIRATION_TIME_OFFSET = 5 * 60 * 1000;
...
-/** Cloud resource manager URL used to retrieve project information. */
+/**
+ * Cloud resource manager URL used to retrieve project information.
+ *
+ * @deprecated use {@link BaseExternalAccountClient.cloudResourceManagerURL} instead
+ **/
 export const CLOUD_RESOURCE_MANAGER =
   'https://cloudresourcemanager.googleapis.com/v1/projects/';
-/** The workforce audience pattern. */
-const WORKFORCE_AUDIENCE_PATTERN =
```
Comment (lsirac) [inline src/auth/baseexternalclient.ts:60]: Please revert these changes -- the audience is always iam.googleapis.com.

File: src/auth/downscopedclient.ts [hunk 4]
```diff
@@ -75,6 +73,13 @@ export interface CredentialAccessBoundary {
   accessBoundary: {
     accessBoundaryRules: AccessBoundaryRule[];
   };
+  /**
+   * An optional STS access token exchange endpoint.
+   *
+   * @example
+   * 'https://sts.googleapis.com/v1/token'
+   */
+  tokenURL?: string | URL;
```
Comment (lsirac) [inline src/auth/downscopedclient.ts:82]: Any reason to expose this? We don't in other languages.

File: src/auth/oauth2client.ts [hunk 5]
```diff
@@ -460,64 +529,40 @@ export class OAuth2Client extends AuthClient {
...
+      tokenInfoUrl: `https://oauth2.${this.universeDomain}/tokeninfo`,
```
Comment (lsirac) [inline src/auth/oauth2client.ts:534]: I am not sure if these are in scope?

File: src/auth/baseexternalclient.ts (hunk 3 above)
Comment (danielbankhead) [reply src/auth/baseexternalclient.ts:60]: Revert PR: https://github.com/googleapis/google-auth-library-nodejs/pull/1740

File: src/auth/downscopedclient.ts (hunk 4 above)
Comment (danielbankhead) [reply src/auth/downscopedclient.ts:82]: Opened a PR to revert for consistency: https://github.com/googleapis/google-auth-library-nodejs/pull/1744

File: src/auth/oauth2client.ts (hunk 5 above)
Comment (danielbankhead) [reply src/auth/oauth2client.ts:534]: Opened a PR to revert: https://github.com/googleapis/google-auth-library-nodejs/pull/1744
